packages/back-end/types/organization.d.ts

```diff
@@ -274,3 +275,12 @@ export type NamespaceUsage = Record<
     end: number;
   }[]
 >;
+
+export type ReqContext = {
+  org: OrganizationInterface;
+  userId: string;
+  email: string;
+  environments: string[];
+  userName: string;
+  readAccessFilter: ReadAccessFilter;
+};
```

Comment:
To follow up on the 'ApiReqContext' comment, we could leverage utility types to ensure the two interfaces stay in sync:

```
// req.user is not always guaranteed within API requests
export type ApiReqContext = Omit<ReqContext, 'userId' | 'email' | 'userName'> & {
  userId?: string;
  email?: string;
  userName?: string;
};
```